src/test/ui/allocator/no_std-alloc-error-handler-custom.rs

```diff
@@ -84,6 +84,9 @@ fn panic(panic_info: &core::panic::PanicInfo) -> ! {
     }
 }
 
+#[lang = "eh_personality"]
+extern fn rust_eh_personality() {}
```

Comment:
This shouldn't be necessary for `-Cpanic=abort`.

Reply:
That's what I thought as well!

I spoke with @/alexcrichton about this in a DM, and he provided some insight that I'll share here. I believe `core` and `alloc` are being compiled with `panic=unwind`, which means that they _may_ refer to `rust_eh_personality` unless LLVM can prove the contents won't unwind.

I tried to find some links to point to where that panic runtime is set, but I'm rather unfamiliar with all of the internals of `compiletest` and the `rustc` test suite in general. Ideally, these tests could be linked against a `panic=abort` version of `alloc`, but I got the impression that this would require a substantial effort.

Assuming I'm not mistaken in any of the above paragraphs, maybe it would be helpful to attach a comment explaining this background information to these `#[lang = ..]` lines. What do you think, @bjorn3?

Reply:
Ah, yeah didn't think about the fact that libcore amd liballoc are compiled with `-Cpanic=unwind`.

Reply:
A comment could be useful to help others who may be confused.

Reply:
A comment has been affixed to each of these `extern fn rust_eh_personality` functions. I've amended the commit where these changed, which can now be found at 4dba3a1 :slightly_smiling_face: